packages/store/schematics/src/ng-add/ng-add.factory.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  Tree,
+  chain,
+  noop
+} from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import * as ts from '@schematics/angular/third_party/github.com/Microsoft/TypeScript/lib/typescript';
+import { addImportToModule } from '@schematics/angular/utility/ast-utils';
+import { InsertChange } from '@schematics/angular/utility/change';
+import {
+  NodeDependencyType,
+  addPackageJsonDependency,
+  getPackageJsonDependency
+} from '@schematics/angular/utility/dependencies';
+import { getAppModulePath } from '@schematics/angular/utility/ng-ast-utils';
+
+import { LIBRARIES } from '../utils/common/lib.config';
+import { getProject } from '../utils/project';
+
+import { NgxsPackageSchema } from './ng-add.schema';
+
+const versions = require('./../utils/versions.json');
+
+export function ngAdd(options: NgxsPackageSchema): Rule {
+  return chain([
+    addNgxsPackageToPackageJson(options),
+    addDeclarationToNgModule(options),
+    options.skipInstall ? noop() : runNpmPackageInstall()
+  ]);
+}
+
+function addNgxsPackageToPackageJson(options: NgxsPackageSchema): Rule {
+  return (host: Tree, context: SchematicContext) => {
+    const ngxsStoreVersion: string = versions['@ngxs/store'];
+    if (!ngxsStoreVersion) {
+      throw new SchematicsException('Could not resolve the version of "@ngxs/store"');
+    }
+
+    Object.values(LIBRARIES)
+      .filter(lib => options.packages?.includes(lib))
```

Comment:
If the `packages` option is reflected in the CLI, I was wondering if renaming it to `plugins` will be more consistent with this https://www.ngxs.io/plugins